thirdparty/nlohmann_json/nlohmann_json/CMakeLists.txt

```diff
@@ -1,7 +1,14 @@
 add_library(nlohmann_json INTERFACE)
 target_sources(nlohmann_json INTERFACE
-  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/single_include/nlohmann/json.hpp>
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/single_include/upstream/nlohmann/json.hpp>
 )
+
+target_sources(nlohmann_json INTERFACE
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/single_include/nlohmann_json.hpp>
+)
+
+target_compile_options(nlohmann_json INTERFACE $<$<AND:$<COMPILE_LANGUAGE:CXX>,$<CXX_COMPILER_ID:Intel>>:-Wno-all -Wno-error>)
```

Comment:
@chuckatkins Since Intel compiler does not like warning pragmas I had to disable all the warnings for TU including nlohmann_json, which are a handful of them.